log/logtest/recorder_test.go

```diff
@@ -0,0 +1,156 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package logtest
+
+import (
+	"context"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"go.opentelemetry.io/otel/log"
+)
+
+func TestRecorderLogger(t *testing.T) {
+	for _, tt := range []struct {
+		name    string
+		options []Option
+
+		loggerName    string
+		loggerOptions []log.LoggerOption
+
+		expectedLogger log.Logger
```

Comment:
Could we use `want` term instead of `expected`?

References:
- https://go.dev/wiki/CodeReviewComments#useful-test-failures
- https://google.github.io/styleguide/go/decisions#got-before-want